filter/series_by_tag.go

```diff
@@ -114,42 +116,95 @@ func createMatcher(spec TagSpec) func(string, map[string]string) bool {
 	}
 }
 
+type MatchingFunc func(string, map[string]string) bool
+
 // SeriesByTagPatternIndex helps to index the seriesByTag patterns and allows to match them by metric
 type SeriesByTagPatternIndex struct {
-	patternToMatcher map[string]func(string, map[string]string) bool
+	namesPatternIndex      *PatternIndex
+	nameToPatternMatchers  map[string]map[string]MatchingFunc
+	specialPatternMatchers map[string]MatchingFunc
 }
 
 // NewSeriesByTagPatternIndex creates new SeriesByTagPatternIndex using seriesByTag patterns and parsed specs comes from ParseSeriesByTag
-func NewSeriesByTagPatternIndex(tagSpecsByPattern map[string][]TagSpec) *SeriesByTagPatternIndex {
-	patternToMatcher := make(map[string]func(string, map[string]string) bool)
+func NewSeriesByTagPatternIndex(logger moira.Logger, tagSpecsByPattern map[string][]TagSpec) *SeriesByTagPatternIndex {
+	names := make(map[string]bool)
+	nameToPatternMatchers := make(map[string]map[string]MatchingFunc)
+	specialPatternsToMatcher := make(map[string]MatchingFunc)
```

Comment:
Would it make more descriptive to call it `wildcardPatternsToMatcher` or something like that?

Reply:
It has changed to withoutStrictNamePatternMatchers